docs/code/EntryPoints.md

```diff
@@ -0,0 +1,188 @@
+﻿# Overview
+
+An 'entry point', is a representation of a ML.Net type in json format and it is used to serialize and deserialize an ML.Net type in JSON. 
+It is also one of the ways ML.Net uses to deserialize experiments, and the recommended way to interface with other languages. 
+In terms defining experiments w.r.t entry points, experiments are entry points DAGs, and respectively, entry points are experiment graph nodes.
+That's why through the documentaiton, we also refer to them as 'entry points nodes'.
+The graph 'variables', the various values of the experiemnt graph json properties serve to describe the relationship between the entry point nodes. 
+The 'variables' are therefore the edges of the DAG. 
+
+All of ML.Net entry points are described by their manifest. The manifest is another json object that documents and describes the structure of an entry points. 
+Manifests are referenced to understand what an entry point does, and how it should be constructed, in a graph.  
+
+This document briefly describes the structure of the entry points, the structure of an entry point manifest, and mentions the ML.Net classes that help construct an entry point
+graph.
+
+## `EntryPoint manifest - the definition of an entry point`
+
+An example of an entry point manifest object, specifically for the MissingValueIndicator transform, is:
+
+```javascript
+    {
+      "Name": "Transforms.MissingValueIndicator",
+      "Desc": "Create a boolean output column with the same number of slots as the input column, where the output value is true if the value in the input column is missing.",
+      "FriendlyName": "NA Indicator Transform",
+      "ShortName": "NAInd",
+      "Inputs": [
+        {
+          "Name": "Column",
+          "Type": {
+            "Kind": "Array",
+            "ItemType": {
+              "Kind": "Struct",
+              "Fields": [
+                {
+                  "Name": "Name",
+                  "Type": "String",
+                  "Desc": "Name of the new column",
+                  "Aliases": [
+                    "name"
+                  ],
+                  "Required": false,
+                  "SortOrder": 150.0,
+                  "IsNullable": false,
+                  "Default": null
+                },
+                {
+                  "Name": "Source",
+                  "Type": "String",
+                  "Desc": "Name of the source column",
+                  "Aliases": [
+                    "src"
+                  ],
+                  "Required": false,
+                  "SortOrder": 150.0,
```

Comment:
>"SortOrder": 150.0, [](start = 18, length = 19)

These are kind of poor examples... `SortOrder` is identical between the two properties here, and in the enclosing scope they are also two properties, etc.

Reply:
leaving the 150 sort order intact, since it seems to be the de fact (not sure if intentional, though) default for advanced properties. 

Updating the transform used for the example to a better one. 

---
In reply to: [192868923](https://github.com/dotnet/machinelearning/pull/295#discussion_r192868923) [](ancestors = 192868923)